src/app/types.ts

```diff
@@ -1,6 +1,8 @@
 export enum NodeType {
   FOLDER = "FOLDER",
   FILE = "FILE",
+  FULL_FOLDER = "FULL_FOLDER",
+  TEXT_FILE = "TEXT_FILE"
```

Comment:
I think that this Enum is already complete. As there is only ever two types for a node : A File and a Folder.